pkg/controller/provider/container/openstack/model.go

```diff
@@ -164,17 +177,89 @@ func (r *ProjectAdapter) GetUpdates(ctx *Context, lastSync time.Time) (updates [
 	return
 }
 
+type ImageAdapter struct {
+}
+
+func (r *ImageAdapter) List(ctx *Context) (itr fb.Iterator, err error) {
+	opts := &ImageListOpts{}
+	imageList := []Image{}
+	err = ctx.client.list(&imageList, opts)
+	if err != nil {
+		return
+	}
+	list := fb.NewList()
+	for _, image := range imageList {
+		m := &model.Image{
+			Base: model.Base{ID: image.ID},
+		}
+		image.ApplyTo(m)
+		list.Append(m)
+	}
+	itr = list.Iter()
+
+	return
+}
+
+func (r *ImageAdapter) GetUpdates(ctx *Context, lastSync time.Time) (updates []Updater, err error) {
+	opts := &ImageListOpts{}
+	opts.setUpdateAtQueryFilterGTE(lastSync)
+	imageList := []Image{}
+	err = ctx.client.list(&imageList, opts)
+	if err != nil {
+		return
+	}
+	for _, image := range imageList {
+		switch image.Status {
+		case ImageStatusDeleted, ImageStatusPendingDelete:
+			updater := func(tx *libmodel.Tx) (err error) {
+				m := &model.Image{
+					Base: model.Base{ID: image.ID},
+				}
+				image.ApplyTo(m)
+				err = tx.Delete(m)
+				return
+			}
+			updates = append(updates, updater)
+
+		default:
+			updater := func(tx *libmodel.Tx) (err error) {
+				m := &model.Image{
+					Base: model.Base{ID: image.ID},
+				}
+				err = tx.Get(m)
+				if err != nil {
+					if errors.Is(err, &NotFound{}) {
```

Comment:
should be
```suggestion
					if errors.Is(err, libmodel.NotFound) {
```
no? (same for the other models)

Reply:
You are right, I guess this is part was copied from another provider. I will change it in the next push